code/linalg.c

```diff
@@ -183,6 +183,21 @@ static mp_obj_t linalg_dot(mp_obj_t _m1, mp_obj_t _m2) {
 
 MP_DEFINE_CONST_FUN_OBJ_2(linalg_dot_obj, linalg_dot);
 
+static mp_obj_t linalg_norm(mp_obj_t _x) {
+    if (!MP_OBJ_IS_TYPE(_x, &ulab_ndarray_type)) {
+        mp_raise_TypeError(translate("argument must be ndarray"));
+    }
+    ndarray_obj_t *x = MP_OBJ_TO_PTR(_x);
+    mp_float_t dot = 0.0, v;
+    for (size_t pos=0; pos < x->array->len; pos++) {
+        v = ndarray_get_float_value(x->array->items, x->array->typecode, pos);
+        dot += v*v;
```

Comment:
Check out, whether `ord` with numerical values makes sense.

Reply:
What do to with numerical 'ord' ? From [numpy.linalg.norm](https://numpy.org/doc/stable/reference/generated/numpy.linalg.norm.html) :
- for vectors, ord != 0 yields `sum(abs(x)**ord)**(1./ord)`;
- for matrices, it is a mess (in NumPy), with many cases...

Reply:
That's right. Let's leave it, then.